### PR TITLE
Converting ios objective-c to mac os swift

### PR DESCRIPTION
**Pull request by thierryH91200**
It is not always easy to write a program in swift, while the examples
are in objective-c
May be that there will be fewer questions

### CODE REVIEW

**Comment by petester42** [comment]
Awesome stuff! 👏

**Comment by thierryH91200** [comment]
I hope it will not fall into oblivion

File: ChartsDemo-OSX/ChartsDemo-OSX/AppDelegate.swift [hunk 1]
```diff
@@ -1,7 +1,25 @@
+//
+//  AppDelegate.swift
+//  graphCompta
+//
+//  Created by thierryH24A on 31/01/2017.
```
Comment (liuxuan30) [inline ChartsDemo-OSX/ChartsDemo-OSX/AppDelegate.swift]: maybe follow the file template and add your contribution

File: ChartsDemo-OSX/ChartsDemo-OSX/AppDelegate.swift [hunk 2]
```diff
@@ -1,7 +1,25 @@
+//
+//  AppDelegate.swift
+//  graphCompta
```
Comment (liuxuan30) [inline ChartsDemo-OSX/ChartsDemo-OSX/AppDelegate.swift]: maybe follow the universal file template to give credits and add yours.

File: ChartsDemo-OSX/ChartsDemo-OSX/AppDelegate.swift [hunk 3]
```diff
@@ -1,7 +1,25 @@
...
+//  Copyright © 2017 thierryH24A. All rights reserved.
+//
+
 import Cocoa
 
 @NSApplicationMain
-class AppDelegate: NSObject, NSApplicationDelegate
-{
-}
+class AppDelegate: NSObject, NSApplicationDelegate {
```
Comment (liuxuan30) [inline ChartsDemo-OSX/ChartsDemo-OSX/AppDelegate.swift]: `{` and `}` should use separate lines, double check others.

File: Source/Charts/Components/AxisBase.swift [hunk 4]
```diff
@@ -41,6 +41,25 @@ open class AxisBase: ComponentBase
     open var drawGridLinesEnabled = true
     open var drawAxisLineEnabled = true
     
+    open var logarithmicEnabled   = false
```
Comment (liuxuan30) [inline Source/Charts/Components/AxisBase.swift]: "log" curve feature gets mixed in this PR, which should be removed.

**Comment by liuxuan30** [comment]
it's a big effort, thank you! Add some comments inline. I can't walk through the controllers one by one though.

**Pull request closed without merging**